### DISCUSSION TITLE
Get the latest workflow run from GitHub Actions using the Octokit API

### DISCUSSION THREAD

**Discussion by courtneycb** [community/community#25451] [Actions]
<p>I’m using the <a href="https://octokit.github.io/rest.js/v18" rel="noopener nofollow ugc">Octokit API</a> to request a specific workflow run. I can see in the <a href="https://octokit.github.io/rest.js/v18#actions-get-workflow-run" rel="noopener nofollow ugc">documentation</a> that I can request a specific workflow run by providing a workflow run ID. However I would like to always request the  <strong>latest</strong> workflow run. How do I achieve this if I don’t know what the ID will be?</p>
<p>E.g. the value of  <code>workflow-run-latest</code>  here would always be changing.</p>
<pre><code class="lang-auto">octokit.actions.getWorkflowRun({
    owner: owner,
    repo: repo-name,
    run_id: workflow-run-latest,
});
</code></pre>
<p>Thanks!</p>

**Comment by dawidd6** [comment, marked as answer]
<p>There should probably be something like:</p>
<pre><code class="lang-auto">octokit.actions.listWorkflowRuns({
    owner: owner,
    repo: repo-name,
    workflow_id: "tests.yml",
    per_page: 1,
});
</code></pre>
<p>or similar. This should give you the latest run for given workflow.</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <a href="https://octokit.github.io/rest.js/v18/" target="_blank" rel="noopener nofollow ugc">octokit.github.io</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="" height="">

<h3><a href="https://octokit.github.io/rest.js/v18/" target="_blank" rel="noopener nofollow ugc">octokit/rest.js</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by courtneycb** [comment]
<p>Thanks heaps for your help! Just posting the code snippet that solved my issue in case this helps anyone in the future.</p>
<pre><code class="lang-auto">    var workflowRuns = octokit.actions.listWorkflowRuns({
        owner: owner,
        repo: 'repo-name',
        workflow_file_name: 'file-name.yaml',
        per_page: 1
    });
</code></pre>
